File: SwissO.Droid/OverviewLayout.cs
using Android.Content;
using Android.Util;
using Android.Views;
using Android.Widget;
using System;
using System.Linq;
using static SwissO.Event;
using static SwissO.MyResources;

namespace SwissO.Droid {
    class OverviewLayout : FrameLayout {

        private Event eventt;

        private OverviewAdapter adapter;

        private ImageView expandView;
        private OverviewButton[] btnAlways;
        private TextView[] tvAlways;
        private OverviewButton[] btnExpand;
        private TextView[] tvExpand;
        public OverviewLayout(Context context, IAttributeSet attrs) : base(context, attrs) {
        }

        public void Init(Event e, AppManager appManager, OverviewAdapter adapter) {
            eventt = e;
            this.adapter = adapter;

            TextView tvtitle = (TextView)FindViewById(Resource.Id.overview_item_title);
            TextView tvdate = (TextView)FindViewById(Resource.Id.overview_item_date);
            TextView tvmap = (TextView)FindViewById(Resource.Id.overview_item_map);
            TextView tvclub = (TextView)FindViewById(Resource.Id.overview_item_club);
            TextView tvdeadline = (TextView)FindViewById(Resource.Id.overview_item_deadline);
            TextView tvregion = (TextView)FindViewById(Resource.Id.overview_item_region);
            tvAlways = new TextView[] { tvtitle, tvdate, tvmap, tvclub, tvdeadline };
            tvExpand = new TextView[] { tvregion };

            tvtitle.Text = " " + eventt.Title;
            tvdate.Text = " " + GetDateString(eventt.Date);
            tvmap.Text = " " + eventt.Map;
            tvclub.Text = " " + eventt.Club;
            tvdeadline.Text = " " + GetDateString(eventt.Deadline);
            tvregion.Text = " " + eventt.Region;
            foreach (TextView tv in tvAlways) {
                SetTextViewVisible(tv);
            }
            foreach (TextView tv in tvExpand) {
                SetTextViewVisible(tv);
            }

            (UriArt[] uris, StringResource[] resources) = OverviewManager.GetUrisSorted(eventt);

            btnAlways = new OverviewButton[4];
            btnAlways[0] = (OverviewButton)FindViewById(Resource.Id.overview_item_button1);
            btnAlways[1] = (OverviewButton)FindViewById(Resource.Id.overview_item_button2);
            btnAlways[2] = (OverviewButton)FindViewById(Resource.Id.overview_item_button3);
            btnAlways[3] = (OverviewButton)FindViewById(Resource.Id.overview_item_button4);
            btnExpand = new OverviewButton[4];
            btnExpand[0] = (OverviewButton)FindViewById(Resource.Id.overview_item_button5);
            btnExpand[1] = (OverviewButton)FindViewById(Resource.Id.overview_item_button6);
            btnExpand[2] = (OverviewButton)FindViewById(Resource.Id.overview_item_button7);
            btnExpand[3] = (OverviewButton)FindViewById(Resource.Id.overview_item_button8);

            OverviewButton[] btns = btnAlways.Concat(btnExpand).ToArray();

            int btni = 0;
            int conti = 0;
            while (conti < uris.Length) {
                if (e.GetUri(uris[conti]) != null) {
                    btns[btni].Init(appManager, resources[conti], e, uris[conti]);
                    btni++;
                }
                conti++;
            }
            while (btni < btns.Length) {
                btns[btni].Init(null, 0, null, 0);
                btni++;
            }
            Collapse();
        }

        internal void SetExpandViewClick() {
            expandView = (ImageView)FindViewById(Resource.Id.overview_item_expand);
            expandView.Click += (e, sender) => {
                adapter.ExpandViewClick(this);
            };
        }

        public void Expand() {
            expandView.SetImageResource(Resource.Drawable.ic_arrow_collapse);
            foreach (OverviewButton btn in btnExpand) {
                btn.SetVisible();
            }
            foreach (TextView tv in tvExpand) {
                SetTextViewVisible(tv);
            }
        }

        public void Collapse() {
            expandView.SetImageResource(Resource.Drawable.ic_arrow_expand);
            foreach (OverviewButton btn in btnExpand) {
                btn.Visibility = ViewStates.Gone;
            }
            foreach (TextView tv in tvExpand) {
                tv.Visibility = ViewStates.Gone;
            }
        }

        private string GetDateString(DateTime value) {
            return DateTime.Compare(value, DateTime.MinValue) == 0 ? "" : value.ToLongDateString();
        }

        private void SetTextViewVisible(TextView tv) {
            tv.Visibility = string.IsNullOrWhiteSpace(tv.Text) ? ViewStates.Gone : ViewStates.Visible;
        }
    }
}

File: app/src/main/java/ch/swisso/OverviewLayout.java
package ch.swisso;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

public class OverviewLayout extends FrameLayout {

    public OverviewLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    @SuppressLint("SetTextI18n")
    public void init(@NonNull Event event, MainActivity act) {

        TextView tvtitle = findViewById(R.id.overview_item_title);
        TextView tvdate = findViewById(R.id.overview_item_date);
        TextView tvmap = findViewById(R.id.overview_item_map);
        TextView tvclub = findViewById(R.id.overview_item_club);
        TextView tvdeadline = findViewById(R.id.overview_item_deadline);
        TextView[] textViews = new TextView[]{tvtitle, tvdate, tvmap, tvclub, tvdeadline};

        tvtitle.setText(event.getName());
        tvtitle.setOnClickListener(v -> tvtitle.setSelected(!tvtitle.isSelected()));
        String date = " " + getDateString(event.getBeginDate());
        if (event.getEndDate() != null) {
            date += " - " + getDateString(event.getEndDate());
        }
        tvdate.setText(date);
        tvmap.setText(" " + event.getMap());
        tvclub.setText(" " + event.getClub());
        tvdeadline.setText(" " + getDateString(event.getDeadline()));
        for (TextView tv : textViews) {
            setTextViewVisible(tv);
        }

        boolean over = Helper.getToday().after(event.getEndDate() != null ? event.getEndDate() : event.getBeginDate()) || event.getUri(Event.UriArt.Rangliste) != null;
        boolean deadlinePassed = Helper.getToday().after(event.getDeadline());
        Event.UriArt[] uris;
        int[] resources;
        Event.UriArt startliste = event.getUri(Event.UriArt.Startliste) != null ? Event.UriArt.Startliste : Event.UriArt.Teilnehmerliste;
        int startlistString = event.getUri(Event.UriArt.Startliste) != null ? R.string.startlist : R.string.teilnehmer;
        if (over) {
            uris = new Event.UriArt[]{Event.UriArt.Ausschreibung, startliste, Event.UriArt.Rangliste, Event.UriArt.WKZ};
            resources = new int[]{R.string.ausschreibung, startlistString, R.string.rangliste, R.string.wkz};
        } else if (deadlinePassed) {
            uris = new Event.UriArt[]{Event.UriArt.Ausschreibung, startliste, Event.UriArt.Liveresultate, Event.UriArt.WKZ, Event.UriArt.Kalender};
            resources = new int[]{R.string.ausschreibung, startlistString, R.string.liveresult, R.string.wkz, R.string.kalender};
        } else {
            uris = new Event.UriArt[]{Event.UriArt.Ausschreibung, startliste, Event.UriArt.Liveresultate, Event.UriArt.WKZ, Event.UriArt.Kalender};
            resources = new int[]{R.string.ausschreibung, startlistString, R.string.liveresult, R.string.wkz, R.string.kalender};
        }

        OverviewButton[] buttons = new OverviewButton[8];
        buttons[0] = findViewById(R.id.overview_item_button1);
        buttons[1] = findViewById(R.id.overview_item_button2);
        buttons[2] = findViewById(R.id.overview_item_button3);
        buttons[3] = findViewById(R.id.overview_item_button4);
        buttons[4] = findViewById(R.id.overview_item_button5);
        buttons[5] = findViewById(R.id.overview_item_button6);
        buttons[6] = findViewById(R.id.overview_item_button7);
        buttons[7] = findViewById(R.id.overview_item_button8);

        int btni = 0;
        int conti = 0;
        while (conti < uris.length) {
            if (event.getUri(uris[conti]) != null) {
                buttons[btni].init(act, resources[conti], event, uris[conti]);
                btni++;
            }
            conti++;
        }
        while (btni < buttons.length) {
            buttons[btni].setVisibility(GONE);
            btni++;
        }

        OnClickListener favOnClickListener = v -> {
            event.toggleFavorit();
            ContentValues values = new ContentValues();
            values.put(SQLiteHelper.COLUMN_FAVORIT, event.isFavorit() ? 1 : 0);
            act.getDaten().updateEvent(values, event.getId());
            findViewById(R.id.overview_item_fav_checkbox_enabled).setVisibility(event.isFavorit() ? VISIBLE : INVISIBLE);
            findViewById(R.id.overview_item_fav_checkbox_disabled).setVisibility(event.isFavorit() ? INVISIBLE : VISIBLE);
        };

        findViewById(R.id.overview_item_fav_checkbox_enabled).setOnClickListener(favOnClickListener);
        findViewById(R.id.overview_item_fav_checkbox_disabled).setOnClickListener(favOnClickListener);
        findViewById(R.id.overview_item_fav_checkbox_enabled).setVisibility(event.isFavorit() ? VISIBLE : INVISIBLE);
        findViewById(R.id.overview_item_fav_checkbox_disabled).setVisibility(event.isFavorit() ? INVISIBLE : VISIBLE);

        setOnClickListener(v -> act.openEventDetails(event, Event.UriArt.Details));
    }

    @NonNull
    private String getDateString(Date date) {
        return date == null ? "" : SimpleDateFormat.getDateInstance(DateFormat.LONG).format(date);
    }

    private void setTextViewVisible(@NonNull TextView tv) {
        String s = tv.getText().toString().trim();
        tv.setVisibility(s.isEmpty() || s.equals("null") ? GONE : VISIBLE);
    }
}
